src/transformStateWithClones.js

```diff
@@ -7,7 +7,51 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = Object.assign({}, state);
+  const allSteps = [];
+
+  function doAction(modState, types, action) {
+    switch (types) {
+      case 'addProperties':
+        addProperties(modState, action.extraData);
+        allSteps.push({ ...modState });
```

Comment:
you can push only one time after switch/case

Reply:
if I understand, I fixed it !)